src/Entity/IdeasWorkshop/BaseComment.php

```diff
@@ -76,4 +63,14 @@ public static function getVisibleStatuses(): array
     {
         return ThreadCommentStatusEnum::VISIBLE_STATUSES;
     }
+
+    public function approve(): void
```

Comment:
What about adding these two methods in the `EntityThreadCommentStatusTrait` ? and set directly the property like : 

```php 
public function approve(): void
{
    $this->status = ThreadCommentStatusEnum::APPROVED;
}